### PR TITLE
Remove implicit copy due to conversion from cudf::size_type and size_t

### PR DESCRIPTION
**Pull request by robertmaynard**
Detected when compiling with gcc-11 by the new `range-loop-construct` warning

### CODE REVIEW

**Review by vuule — APPROVED** [review]

**Review by bdice — APPROVED** [review]
That's a nice fix. LGTM.

**Review by devavret — APPROVED** [review]
What was being copied?

**Comment by robertmaynard** [comment]
> What was being copied?

The line `for (const size_t& stripe_idx : user_specified_stripes[src_file_idx]) ` has an implicit copy since `user_specified_stripes` is a vector of `cudf::size_t` which are 32bit and `size_t` is 64bit. 

So while it appears we are using the values inside `user_specified_stripes` we are actually const ref binding to temporaries

**Comment by robertmaynard** [comment]
rerun tests

File: cpp/src/io/orc/aggregate_orc_metadata.cpp [hunk 1]
```diff
@@ -171,8 +171,9 @@ std::vector<metadata::stripe_source_mapping> aggregate_orc_metadata::select_stri
 
       // Coalesce stripe info at the source file later since that makes downstream processing much
       // easier in impl::read
-      for (const size_t& stripe_idx : user_specified_stripes[src_file_idx]) {
-        CUDF_EXPECTS(stripe_idx < per_file_metadata[src_file_idx].ff.stripes.size(),
+      for (const auto& stripe_idx : user_specified_stripes[src_file_idx]) {
```
Comment (ttnghia) [inline cpp/src/io/orc/aggregate_orc_metadata.cpp:174]: Just small nit:
```suggestion
      for (const auto stripe_idx : user_specified_stripes[src_file_idx]) {
```

**Review by ttnghia — APPROVED** [review]

File: cpp/src/io/orc/aggregate_orc_metadata.cpp (hunk 1 above)
Comment (bdice) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:174]: @ttnghia Can you explain why you're suggesting this change?

**Comment by robertmaynard** [comment]
@gpucibot merge

File: cpp/src/io/orc/aggregate_orc_metadata.cpp (hunk 1 above)
Comment (ttnghia) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:174]: Sorry for late reply (I didn't see this).
A reference is similar to pointer, which is 64 bits. If you use reference for 32 bits number, that's kind of inefficient. Although the compiler can optimize this and generate the same code (for references and without references), it is still logically inefficient.

**Pull request merged**